nixos/modules/services/system/dbus.nix

```diff
@@ -122,6 +141,30 @@ in
 
     systemd.services.dbus.restartTriggers = [ configDir ];
 
+    systemd.user = mkIf config.systemd.user.dbus.enable {
+      services.dbus = {
+        description = "D-Bus User Message Bus";
+        requires = [ "dbus.socket" ];
```

Comment:
Should it also be `after = [ "dbus.socket" ]` ? Because the socket sets `DBUS_SESSION_BUS_ADDRESS`, there might be some race condition with which dbus-daemon starts disregarding the dbus.socket.